includes/event/event-repository-interface.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Wporg\TranslationEvents\Event;
+
+use Exception;
+use Throwable;
+
+class EventNotFound extends Exception {
+	public function __construct( Throwable $previous = null ) {
+		parent::__construct( 'Event not found', 0, $previous );
+	}
+}
+
+class CreateEventFailed extends Exception {}
+class UpdateEventFailed extends Exception {}
+
+interface Event_Repository_Interface {
+	/**
+	 * @throws CreateEventFailed
+	 */
+	public function create_event( Event $event ): void;
```

Comment:
Could we use the same approach that we use in with the posts: `insert_event( Event $event )` to create and update the event and `update_event( Event $event )` to update the event. 

I think it will be interesting that both methods will return `int|WP_Error`, where `int` is the event ID.

And I think we need a method to delete the events: `delete_event( int $eventid, bool $force_delete = false ): Event|false`: 
- Parameters:
  - $eventid: the id of the event to delete.
  - $force_delete: delete the event if it has translations. false by default.
- Returned value: Event|false Event data on success, false on failure.

[wp_insert_post()](https://developer.wordpress.org/reference/functions/wp_insert_post/) and [wp_update_post()](https://developer.wordpress.org/reference/functions/wp_update_post/) documentation.

Reply:
I have now implemented these suggestions.

Concerning `delete_event()`, for now the function doesn't take the `$force_delete` argument. I didn't do it here because I'm not sure it's something that belongs in the repository as it feels like it's more a business logic thing, so it would be the calling code that needs to check if the event has stats before calling `delete_event()`. That's already the case today, [here](https://github.com/WordPress/wporg-gp-translation-events/blob/trunk/wporg-gp-translation-events.php#L317-L326).

Currently (in trunk) we're not yet deleting stats when the event is deleted, and I would prefer to not add any features in this PR, so that would be another reason to not add the `$force_delete` flag yet. 

If this makes sense to you, I can create a new issue for deleting stats when the event is deleted.